tests/unit/constraints/test_tabular.py

```diff
@@ -1274,3 +1274,174 @@ def test_reverse_transform(self):
             'c': [5, 7, 9]
         })
         pd.testing.assert_frame_equal(expected_out, out)
+
+
+class TestRounding():
+
+    def test___init__(self):
+        """Test the ``Rounding.__init__`` method.
+
+        It is expected to create a new Constraint instance
+        and set the rounding args.
+
+        Input:
+        - columns = ['b', 'c']
+        - digits = 2
+        """
+        # Setup
+        columns = ['b', 'c']
+        digits = 2
+
+        # Run
+        instance = Rounding(columns=columns, digits=digits)
+
+        # Assert
+        assert instance._columns == columns
+        assert instance._digits == digits
+
+    def test_is_valid_valid(self):
+        """Test the ``Rounding.is_valid`` method for a valid data.
+
+        If the data is within threshold of the desired rounding, the result
+        is a series of ``True`` values.
+
+        Input:
+        - Table data with desired decimal places (pandas.DataFrame)
+        Output:
+        - Series of ``True`` values (pandas.Series)
+        """
+        # Setup
+        columns = ['b', 'c']
+        digits = 2
+        tolerance = 1e-3
+        instance = Rounding(columns=columns, digits=digits, tolerance=tolerance)
+
+        # Run
+        table_data = pd.DataFrame({
+            'a': [1, 2, 3],
+            'b': [4.12, 5.51, 6.901],
+            'c': [5.310, 7.12, 9.120]
+        })
+        out = instance.is_valid(table_data)
+
+        # Assert
+        expected_out = pd.Series([True, True, True])
+        pd.testing.assert_series_equal(expected_out, out)
+
+    def test_is_valid_negative_digit_valid(self):
```

Comment:
I think we could merge some of the `is_valid` tests to cover multiple cases at once.
I would create 3 tests: digits = 0, digits > 0, digits < 0

And each one of the can have:
- 1 Invalid row
- 1 row that is exactly valid
- 1 row that is slightly over the rounded level but still valid (within tolerance)
- 1 row that is slightly below the rounded level but still valid (within tolerance)

Also, at the column level, each test should have:
- Multiple number of rounded columns (make 1 test with 1 column, the others with 2 or more, to cover all cases)
- At least 1 numerical column that is not being rounded, which should have a deeper rounding level to make sure that it is not being rounded
- At least 1 non-numerical column

Finally, all the test cases should have random Null values here and there to ensure that they do not break things